
**Author**: Aaryan Samanta

**Organization**: Legend College Preparatory

**Date**: 2025

**Title**: Exploring the World Happiness Report Dataset

**Version**: 1.0

**Type**: Source Code

**Adaptation details**: Based on classroom exercises

**Description**: The goal of this exercise is to help students work with datasets in Python, focusing on exploring their structure and understanding how the data is organized.



---


Developed as part of the AI Internship at Legend College Preparatory.
Please note that it is a violation of school policy to copy and use this code without proper attribution and credit acknowledgement.
Failing to do so can constitute plagiarism, even with small code snippets.

In [ ]:
from google.colab import files
uploaded = files.upload()
import pandas as pd

# Load the World Happiness Report dataset
url = "https://www.kaggle.com/datasets/ajaypalsinghlo/world-happiness-report-2021"
data = pd.read_csv("world-happiness-report-2021.csv")

# Display the first 5 rows
print(data.head())

# Print the dataset shape
print(f"The dataset has {data.shape[0]} rows and {data.shape[1]} columns.")

# Show a summary of the dataset
print(data.info())

# Show statistical details of numerical columns
print(data.describe())


Saving world-happiness-report-2021.csv to world-happiness-report-2021 (1).csv
  Country name Regional indicator  Ladder score  \
0      Finland     Western Europe         7.842   
1      Denmark     Western Europe         7.620   
2  Switzerland     Western Europe         7.571   
3      Iceland     Western Europe         7.554   
4  Netherlands     Western Europe         7.464   

   Standard error of ladder score  upperwhisker  lowerwhisker  \
0                           0.032         7.904         7.780   
1                           0.035         7.687         7.552   
2                           0.036         7.643         7.500   
3                           0.059         7.670         7.438   
4                           0.027         7.518         7.410   

   Logged GDP per capita  Social support  Healthy life expectancy  \
0                 10.775           0.954                     72.0   
1                 10.933           0.954                     72.7   
2                

The World Happiness Report 2021 dataset has 149 rows and 20 columns, for each of the 149 countries and their respective indicators measuring happiness. Using the function .info() , I found out that the dataset has 18 numeric columns and 2 categorical columns. The 2 categorical columns, namely “Country” and “year,” have unique values for every row so there are no missing values in this dataset. Therefore, there are no gaps to fill in the data, and the data is complete and intact. Using the function .describe() provided the key statistics, mean, min, max, and quartile of each of the numeric columns and provided concrete insight into how countries generally performed based on one of the scores.
I found it really interesting that Finland is listed as the happiest country with a ladder score of 7.842 while the ladder score for the lowest score was 2.523, which indicates a huge gap in overall happiness globally. Also, healthy life expectancy varied in this data from about 48 years to almost 77 years, showing a level of variation between nations and places in terms of healthy healthcare and living conditions. Another interesting surprise I found from my exploration of the data is that the “Generosity” column had negative values, indicating that generosity was being evaluated; either by wealth or perception as a measure of generosity rather than strictly a positive number.